File: javaDesign/src/model/Problem.java
package model;

import java.time.LocalDateTime;
import java.util.List;

/**
 * @author Abhishek V S
 **/
public class Problem {
    private String Title;
    private String Difficulty;
    private String Url;
    private Submissions submissions;

    public Problem(String title, String difficulty, String url) {
        Title = title;
        Difficulty = difficulty;
        Url = url;
    }

    public String getTitle() {
        return Title;
    }

    public void setTitle(String title) {
        Title = title;
    }

    public String getDifficulty() {
        return Difficulty;
    }

    public void setDifficulty(String difficulty) {
        Difficulty = difficulty;
    }

    public String getUrl() {
        return Url;
    }

    public void setUrl(String url) {
        Url = url;
    }

    public Submissions getSubmissions() {
        return submissions;
    }

    public void setSubmissions(LocalDateTime date) {
        submissions.setDates(date);
    }
}

File: extension/models/Problem.js
class Problem{
      constructor(title,difficulty,url) {
            this.title = title;
            this.difficulty = difficulty;
            this.url = url;
            this.submissions=[]
        }
        getTitle(){

            return this.title;
        }
        setTitle(title) {
            this.title = title;
        }

        addSubmission(submission) {
        let check=false;
        for(let i of this.submissions){
            if(i.getSid()==submission.getSid()){
                check=true;
                break;
            }
        }
        if(!check)this.submissions.push(submission);
        }
        getUrl(){
         return this.url;
        }

        getDifficulty(){
         return this.difficulty;
        }
        static fromJson(json){
            const p=new Problem(json.title,
            json.difficulty,
            json.url,
            );

            for(const submission of (json.submissions || [] )){
                p.addSubmission(Submission.fromJson(submission));
            }

            return p;
        }

}